Add getsBulk method to MemcachedClient

Jcparkyn opened the pull request:
Closes #50.

Adds a `getsBulk` method (bulk get with CAS identifiers), based on the implementation of `getBulk`.

IMO there's no real need for the duplication of `MultiGetsOperationImpl` and `MultiGetOperationImpl` (they could both be implemented with `MultiGetsOperationImpl` and `GetsOperation.Callback` with no performance loss), but I kept it for consistency and to avoid modifying existing code. Happy to change if requested.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

jeremyprime commented on src/main/java/net/spy/memcached/MemcachedClient.java:
```diff
@@ -1619,6 +1619,111 @@ public void complete() {
+   * @throws IllegalStateException in the rare circumstance where queue is too
+   *           full to accept any more requests
+   */
+  public <T> BulkFuture<Map<String, CASValue<T>>> asyncGetsBulk(Iterator<String> keyIter,
```
nit: Can we add `@Override` to the new `asyncGetsBulk`/`getsBulk` methods to be consistent with existing methods in this class.

jeremyprime commented on src/main/java/net/spy/memcached/MemcachedConnection.java:
```diff
@@ -1327,6 +1328,11 @@ public void redistributeOperation(Operation op) {
       }
+    } else if (op instanceof MultiGetsOperationImpl) {
+      for (String key : ((MultiGetsOperationImpl) op).getRetryKeys()) {
+        addOperation(key, opFact.get(key,
```
I believe this should be `gets`/`GetsOperation`, not `get`/`GetOperation`:
```
        addOperation(key, opFact.gets(key,
          (GetsOperation.Callback) op.getCallback()));
```

Jcparkyn commented:
Thanks @jeremyprime and @affonsov for review! I've fixed those issues & added a test, and rebased onto master since this branch has just turned one year old 😄
